### DISCUSSION TITLE
Any suggestions are welcome

### DISCUSSION THREAD

**Discussion by giacbli** [giacbli/DcmStoreService#1] [Ideas]
Hi

**Comment by giacbli** [comment]
Your comment is valuable, it helps me improve the package's features and its ease of use

**Comment by peterUZBR** [comment]
Hello Giacomo,

Thank for the nice Dicom SCP software, it is working great.

Would you consider adding the possibility for a flag to enable or disable the .dcm extension for the dicom files in a future release?

We use several softwares that expect .dcm files.

I'm using a rename-tool to add the extension after dumping the dicom-files.
It would be awesome if you could implement this directly in the SCP-software.

Splitting exams using the accession number would also be quite handy.

Also, listening on multiple ports, with each port having a different store location could be a useful add-on.



Best Regards,

Peter

**Reply by giacbli** [reply]
Hello Peter,

I apologize for the delayed, more specific response.

Thank you very much for your appreciation of the DcmStoreService software, I hope it will be useful for you in managing your images.
Yes, thanks for the suggestion; I have addressed your initial request by adding a flag to enable or disable the file extension for DICOM files—going beyond the standard ".dcm" suffix often used by applications to identify the format—while still allowing the user to choose their own extension. You will find this update in release 1.4.1, available on both GitHub and SourceForge.

Thank you also for the excellent idea regarding the package enhancement. I am seriously considering scheduling the implementation of this add-on for a future update. I am currently investigating the best strategy to configure the SCP server to listen on multiple TCP ports, thereby allowing a more flexible management of local storage and remote resources.

For now, as a first approach I would suggest an indirect solution — even if it may seem trivial — by creating multiple instances of the SCP service. To enable listening on multiple ports, you must launch a separate DcmStoreService process for each port. Naturally, you need to duplicate the application folder and modify the configuration file, updating the port value and other parameters as required. You will also need to rename the service instances (e.g., DcmStoreService1, DcmStoreService2, etc.) and potentially modify the scripts (start, stop, query, delete) if you intend to manage them manually.

Best Regards,
Giacomo
